test/ModuleLoader.test.cpp

```diff
@@ -1,18 +1,47 @@
+#include "GameModules/IpDropDown.h"
 #include "ModuleLoader.h"
 #include <gtest/gtest.h>
+#include <memory>
 
-TEST(ModuleLoader, InitializeNoModules)
+
+TEST(ModuleLoader, NoModulesLoaded)
+{
+	ModuleLoader moduleManager;
+	EXPECT_EQ(0, moduleManager.Count());
+	EXPECT_EQ("", moduleManager.GetModuleName(0));
+	EXPECT_FALSE(moduleManager.IsModuleLoaded("Test"));
+	EXPECT_NO_THROW(moduleManager.InitializeModules());
+	EXPECT_NO_THROW(moduleManager.DestroyModules());
+}
+
+TEST(ModuleLoader, NullModulePassed)
+{
+	ModuleLoader moduleManager;
+	std::unique_ptr<GameModule> gameModule;
+	EXPECT_NO_THROW(moduleManager.RegisterModule(gameModule));
+	
+	// A null unique pointer does not count as a loaded module
+	EXPECT_EQ(0, moduleManager.Count());
+}
+
+TEST(ModuleLoader, InternalModulePassed)
 {
-	ModuleLoader moduleLoader;
-	EXPECT_NO_THROW(moduleLoader.LoadModules());
+	ModuleLoader moduleManager;
+	std::unique_ptr<GameModule> ipDropDown = std::make_unique<IPDropDown>();
+
+	moduleManager.RegisterModule(ipDropDown);
+	
+	// Ensure ipDropDown is transfered into moduleManager
+	EXPECT_TRUE(ipDropDown == nullptr);
 	
-	EXPECT_EQ("", moduleLoader.GetModuleName(0));
-	EXPECT_EQ("", moduleLoader.GetModuleName(1));
+	EXPECT_EQ(1, moduleManager.Count());
 
-	EXPECT_FALSE(moduleLoader.IsModuleLoaded(""));
-	EXPECT_FALSE(moduleLoader.IsModuleLoaded("TEST"));
+	// Check module name search is case insensitive
+	EXPECT_TRUE(moduleManager.IsModuleLoaded("IPDROPDOWN"));
```

Comment:
If modules names are case insensitive, we need to be consistent about that treatment. I don't believe internal code always treats them as case insensitive.